### PR TITLE
Fullclose on incoming connection + stream close on send/handle headers

### PR DESCRIPTION
**Pull request by pradovic**
- The both incoming and sending side should wait for each other after adding the node to the peer registry, so we have a consistent situation when initiating protocols.
- handle/send headers stream close per https://github.com/ethersphere/bee/pull/354

### CODE REVIEW

**Comment by acud** [comment]
@pradovic are we proceeding with this for the release?

**Comment by pradovic** [comment]
@acud we wanted to wait a bit on Friday. This should be more correct and cut thr wait in the case of timeout. We canmerve it, but can also wait if we don't want to risk stability. @janos what do you think as well?

**Comment by janos** [comment]
This commit https://github.com/ethersphere/bee/pull/336/commits/9884941c596f794ba66ce885a5c635e662325ec2 was reverted on https://github.com/ethersphere/bee/pull/354.

**Comment by pradovic** [comment]
@janos I have reverted disconnect here as well then, but I left the fullclose sync it should be more correct

File: pkg/p2p/libp2p/libp2p.go [hunk 1]
```diff
@@ -285,6 +285,10 @@ func (s *Service) AddProtocol(p p2p.ProtocolSpec) (err error) {
 			// exchange headers
 			if err := handleHeaders(ss.Headler, stream); err != nil {
 				s.logger.Debugf("handle protocol %s/%s: stream %s: peer %s: handle headers: %v", p.Name, p.Version, ss.Name, overlay, err)
+				s.logger.Errorf("handle protocol %s/%s: peer %s", p.Name, p.Version, overlay)
```
Comment (acud) [inline pkg/p2p/libp2p/libp2p.go:288]: sure we want this on `error`? maybe `debug`?

Comment (pradovic) [reply pkg/p2p/libp2p/libp2p.go:288]: I generally agree, but there will be no message for the user then, cuz we don't return an error here. Maybe warning?

Comment (acud) [reply pkg/p2p/libp2p/libp2p.go:288]: this is not really a user-facing error anyway... not sure what the mere mortal is supposed to make of this sort of error. it is pretty obscure. i'd even go as far as saying just have the debug line above and remove this line. anyway, as you wish

Comment (pradovic) [reply pkg/p2p/libp2p/libp2p.go:288]: Cool, good for me :)

**Pull request merged**